### PR TITLE
Add a popover close button, spell note popover

### PR DESCRIPTION
**Pull request by jazeee**
Ref Popovers #49

I have a popover example working:
image

image

Also, added a close button.
Moving or dragging a popover is not an easy topic in Roll20. We don't really have that kind of control.

image

Converted to pug to allow HTML reuse.
image

### CODE REVIEW

File: ui/components/popover/popover.scss [hunk 1]
```diff
@@ -4,6 +4,9 @@ button.popoverButton {
   border: 1px solid gray;
   border-radius: 100%;
   margin: 0;
+  &.smallerPopoverButton {
+    padding: 2px 6px !important;
```
Comment (jazeee) [inline ui/components/popover/popover.scss]: We have a lot of global styles that are hard to override due to depth of specificity.
Ideally, we whittle those down and reduce the need for `!important`

File: ui/spells/spells.html [hunk 2]
```diff
@@ -93,7 +93,22 @@
...
+        <button name="act_togglepopoverspellnote" class="popoverButton smallerPopoverButton" type="action">
+          ?
+        </button>
+        <!-- The next input puts whether we have a note into the
+                  value property, where CSS can see it -->
+        <input name="attr_visiblepopover" class="popoverController" type="hidden" value="false" />
+        <div class="spellNotePopover popoverContainer">
+          <div class="popover">
+            <!-- Buttons in a popover break stacking order, and pop out of the repeating stack. -->
+            <button name="act_closepopoverspellnotes" class="closePopoverButton" type="action">X</button>
```
Comment (jazeee) [inline ui/spells/spells.html]: Turns out that `position: relative` breaks Roll20 and the button's action functionality.
This button can only act as a top level button, which is fine... We want to close all popovers when we click on this.

File: ui/spells/spells.scss [hunk 3]
```diff
@@ -171,4 +171,16 @@
...
+}
+
+@include popover-control('spellNotePopover');
+div.spellNotePopover {
+  .popover {
+    width: 500px;
+    left: -40vw;
+    top: -40px;
+    label {
+      text-align: left;
```
Comment (jazeee) [inline ui/spells/spells.scss]: Labels default to text-align right at a root style setup. 😢

File: ui/spells/spells.scss [hunk 4]
```diff
@@ -171,4 +171,16 @@
   display: flex;
   flex-direction: row;
   align-items: flex-end;
+}
+
+@include popover-control('spellNotePopover');
+div.spellNotePopover {
+  .popover {
+    width: 500px;
```
Comment (jazeee) [inline ui/spells/spells.scss]: We can make the notes section as large as we want now...

File: ui/scripts/inlineHtml/inlineScripts.js [hunk 5]
```diff
@@ -483,6 +483,29 @@ on('sheet:opened', () => setAttrs({ chosenPopover: 'false' }));
...
+    console.log('*** Close all spell notes ***');
+    ids.forEach((id) => {
+      setAttrs({ [`repeating_arcane_${id}_visiblepopover`]: 'false' });
+    });
+  });
+}
+
+on('sheet:opened', closeAllSpellNotes);
+
+on('clicked:repeating_arcane:togglepopoverspellnote', function () {
```
Comment (jazeee) [inline ui/scripts/inlineHtml/inlineScripts.js]: I need to follow up and apply this for innate and divine.

**Comment by johnlamping** [comment]
This seems good.
For the future, it might be worth experimenting with a way to make the note appear in new space under the spell, when you click on the note icon and  not disappear when you click on a different one. That way, a player can keep open whatever notes they want.
(A lot of my career has been based on being able to see exactly you want, all at once, without having to click around.)

**Review by johnlamping — APPROVED** [review]

**Comment by jazeee** [comment]
I made a few style adjustments.
Next will be to add the backdrop. I have an idea on how to handle this.

**Pull request merged**